code/modules/clothing/clothing.dm

```diff
@@ -798,7 +798,7 @@
 
 /obj/item/clothing/under/proc/detach_accessory(obj/item/clothing/accessory/A, mob/user)
 	accessories -= A
-	A.on_removed(user)
+	A.on_removed(user, src)
```

Comment:
`has_suit` is already the suit to which this accessory is attached to... A horrible name for an object variable since you'd expect a boolean value.

Reply:
Huh, that's not what ie expected from that name. I'll change this when I get home.

Reply:
It doesn't seem like I can use has_suit for on_removed on account of has_suit needs to be set to null in the base proc and thus I must still send src as an argument there.  similarly, since the base proc for on_attached needs obj/item/clothing/under/S there seems to be little use for has_suit there. Is there something im missing that you'd like me to change instead?

Reply:
Gotta do it before you call `..()` in this case. Usually a Destroy like proc (one which deconstructs the object so to say) works from the top downwards. So you do your code first, then the base's implementation. The opposite of an Initialize like proc